### PR TITLE
feat: add Raku module

### PR DESCRIPTION
**Pull request by sergot**
Implement a module to support [Raku](https://www.raku.org/) programming language, it prints the version (and the vm type it's built on) for Raku, for example: `🦋 v6.d-moar`. 🦋 (a butterfly) is Raku's logo called Camelia.

I followed the checklist from [CONTRIBUTING.md](https://github.com/sergot/starship/blob/master/CONTRIBUTING.md#new-module-checklist).

#### Motivation and Context

As a developer who likes to code using Raku language, I'd like to see the current version of raku in the command line.

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [x] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Comment by sergot** [comment]
I'll fix the Dprint issue today.

**Review by andytom — COMMENTED** [review]
LGTM, looks like there are a few gaps in the test coverage can you update the tests to cover these.

**Comment by sergot** [comment]
Hi @andytom , I'd love to but it seems like it's not that easy to do, I went through other modules and haven't found a single one that covers these lines :( 

e.g. https://app.codecov.io/gh/starship/starship/blob/master/src/modules/python.rs

File: src/modules/raku.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let raku_version = context
+                        .exec_cmd("raku", &["-e", "printf q/%s-%s/, $*RAKU.version, $*VM;"])?
+                        .stdout;
```
Comment (davidkna) [inline src/modules/raku.rs:36]: Can you separate the different outputs into different variables, like the elixir module does? I also noticed `raku` has a `--version` flag, which might be faster and exposes information about the runtime version:
```
❯ raku --version
Welcome to Rakudo™ v2021.12.
Implementing the Raku® Programming Language v6.d.
Built on MoarVM version 2021.12.
```

Comment (sergot) [reply src/modules/raku.rs:36]: Good point @davidkna, on it! :)

Comment (sergot) [reply src/modules/raku.rs]: Done

**Review by davidkna — APPROVED** [review]

File: docs/config/README.md [hunk 2]
```diff
@@ -2938,6 +2938,45 @@ any of the following conditions are met:
...
+- The current directory contains a `META6.json` file
+- The current directory contains a `.p6`, `.pm6`, `.raku`, `.rakumod` or `.pod6`
+
+### Options
+
+| Option              | Default                                          | Description                                                               |
+| ------------------- | ------------------------------------------------ | ------------------------------------------------------------------------- |
+| `format`            | `"via [$symbol($version-$vm_version )]($style)"` | The format string for the module.                                         |
+| `version_format`    | `"v${raw}"`                                      | The version format. Available vars are `raw`, `major`, `minor`, & `patch` |
+| `symbol`            | `"🦋 "`                                           | The symbol used before displaying the version of Perl                     |
```
Comment (andytom) [inline docs/config/README.md:2955]: Should this be
```suggestion
| `symbol`            | `"🦋 "`                                           | The symbol used before displaying the version of Raku                     |
```

**Review by andytom — APPROVED** [review]
Found a little typo but other than that all good.

**Comment by davidkna** [comment]
Thanks for the contribution @sergot, and thanks for helping to review this @andytom!

**Pull request merged**